### PR TITLE
PMM-4133 Testing Jenkins Changes

### PR DESCRIPTION
**Pull request by puneet0191**
This PR will allow us to provide live feedback of the changes on API tests, The environment is created for the latest docker tag for this fb, It also adds notification to the build message on ci channel

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-0af1b72
client docker - perconalab/pmm-client-fb:PR-287-0af1b72
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-0af1b72.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-d867dd2
client docker - perconalab/pmm-client-fb:PR-287-d867dd2
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-d867dd2.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-017b79f
client docker - perconalab/pmm-client-fb:PR-287-017b79f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-017b79f.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-338898f
client docker - perconalab/pmm-client-fb:PR-287-338898f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-338898f.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-351a469
client docker - perconalab/pmm-client-fb:PR-287-351a469
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-351a469.tar.gz

**Comment by puneet0191** [comment]
This PR will allow us to provide live feedback of the changes on API tests, The environment is created for the latest docker tag for this fb, It also adds notification to the build message on ci channel

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-2f4b853
client docker - perconalab/pmm-client-fb:PR-287-2f4b853
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-2f4b853.tar.gz

File: Jenkinsfile [hunk 1]
```diff
@@ -23,6 +30,9 @@ pipeline {
                     git submodule foreach --recursive git reset --hard
                     git submodule foreach --recursive git clean -fdx
                     git submodule status
+                    export commit_sha=$(git submodule status | grep 'pmm-managed' | awk -F ' ' '{print $1}')
+                    curl -s https://api.github.com/repos/percona/pmm-managed/commits/${commit_sha} | grep 'name' | awk -F '"' '{print $4}' | head -1 > OWNER
+                    cat OWNER
```
Comment (hors) [inline Jenkinsfile]: Is it really needed?

File: Jenkinsfile [hunk 2]
```diff
@@ -23,6 +30,9 @@ pipeline {
                     git submodule foreach --recursive git reset --hard
                     git submodule foreach --recursive git clean -fdx
                     git submodule status
+                    export commit_sha=$(git submodule status | grep 'pmm-managed' | awk -F ' ' '{print $1}')
+                    curl -s https://api.github.com/repos/percona/pmm-managed/commits/${commit_sha} | grep 'name' | awk -F '"' '{print $4}' | head -1 > OWNER
```
Comment (hors) [inline Jenkinsfile]: Maybe we can get owner using the following command: git --no-pager show -s  --format="%an <%ae>" commit_id

Comment (BupycHuk) [reply Jenkinsfile]: btw, we can get last commit author for submodules using this command 
```git submodule foreach 'git log -1 --pretty=format:"%ae %an"'```

Comment (puneet0191) [reply Jenkinsfile]: I think that would return me github userid for most of the commits, I need names which is returned by the API call, @hors  what you suggest is good if we need info for main repo and not submodule.

File: Jenkinsfile (hunk 1 above)
Comment (puneet0191) [reply Jenkinsfile]: Yeah, Added for debuggin, forgot to remove...thanks I will update it.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-a85d4dd
client docker - perconalab/pmm-client-fb:PR-287-a85d4dd
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-a85d4dd.tar.gz

File: Jenkinsfile [hunk 3]
```diff
@@ -3,6 +3,13 @@ library changelog: false, identifier: 'lib@master', retriever: modernSCM([
     remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
 ]) _
 
+void runAPItests(String DOCKER_VERSION, OWNER) {
+    stagingJob = build job: 'pmm2-api-tests', parameters: [
+        string(name: 'DOCKER_VERSION', value: DOCKER_VERSION),
```
Comment (BupycHuk) [inline Jenkinsfile]: Let's use `DOCKER_IMAGE_VERSION` instead of `DOCKER_VERSION`. It's more accurate.

Comment (hors) [reply Jenkinsfile]: > Let's use `DOCKER_IMAGE_VERSION` instead of `DOCKER_VERSION`. It's more accurate.

+++

Comment (AlekSi) [reply Jenkinsfile]: Or `DOCKER_IMAGE`.

Comment (puneet0191) [reply Jenkinsfile]: Sure, I will update it..and then also update API tests job later, for now just changing the variable name.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-a27191a
client docker - perconalab/pmm-client-fb:PR-287-a27191a
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-a27191a.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-287-a27191a
client docker - perconalab/pmm-client-fb:PR-287-a27191a
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/pmm2-client/pmm2-client-PR-287-a27191a.tar.gz

**Pull request merged**